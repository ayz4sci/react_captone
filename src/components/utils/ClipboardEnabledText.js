import React from 'react';
import PropTypes from 'prop-types';
import {Tooltip} from 'react-tippy';

class ClipboardEnabledText extends React.Component{
    state = {
        copied: false
    }

    copyText = (e) => {
        e.preventDefault();

        this.selectText(e);
        document.execCommand('copy');
        this.setState({ copied: true});
    }

    selectText = (event) => {
        var selection = window.getSelection();            
        var range = document.createRange();
        range.selectNodeContents(event.target);
        selection.removeAllRanges();
        selection.addRange(range);
    }

    render(){
        return (
            <Tooltip
                title="Copied!"
                trigger="click"
                hideDelay="1000"
                inertia="true"
                size="small"
                distance="0"
                ref={tooltip => (this.tooltip = tooltip)}>
                <code onMouseMove={() => this.tooltip.hideTooltip()}>
                    {/* eslint-disable-next-line */}
                    <a href="#" onClick={this.copyText}>{this.props.content}</a>
                </code>
            </Tooltip>

        )
    }
}

ClipboardEnabledText.propTypes = {
    content: PropTypes.string.isRequired
}

export default ClipboardEnabledText;