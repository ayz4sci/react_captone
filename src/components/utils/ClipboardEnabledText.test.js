import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ClipboardEnabledText from './ClipboardEnabledText';

it('renders correctly', () => {
    const props = { 
        content: "lol"
    };

    const wrapper = shallow(<ClipboardEnabledText {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot();
});