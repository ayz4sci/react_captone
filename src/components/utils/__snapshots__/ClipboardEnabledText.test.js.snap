// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Tooltip
  animateFill={true}
  animation="shift"
  arrow={false}
  arrowSize="regular"
  className=""
  delay={0}
  disabled={false}
  distance="0"
  duration={375}
  followCursor={false}
  hideDelay="1000"
  hideDuration={375}
  hideOnClick={true}
  html={null}
  inertia="true"
  interactive={false}
  interactiveBorder={2}
  multiple={false}
  offset={0}
  onHidden={[Function]}
  onHide={[Function]}
  onRequestClose={[Function]}
  onShow={[Function]}
  onShown={[Function]}
  popperOptions={Object {}}
  position="top"
  size="small"
  sticky={false}
  stickyDuration={200}
  style={Object {}}
  theme="dark"
  title="Copied!"
  touchHold={false}
  trigger="click"
  unmountHTMLWhenHide={false}
>
  <code
    onMouseMove={[Function]}
  >
    <a
      href="#"
      onClick={[Function]}
    >
      lol
    </a>
  </code>
</Tooltip>
`;
